clients/java/signalr/src/main/java/HubConnection.java

```diff
@@ -15,21 +15,39 @@
     private CallbackMap handlers = new CallbackMap();
     private HubProtocol protocol;
     private Gson gson = new Gson();
+    private Boolean handshakeReceived = false;
+    private static final String RECORD_SEPARATOR = "\u001e";
     private HubConnectionState connectionState = HubConnectionState.DISCONNECTED;
 
     public HubConnection(String url, Transport transport) {
         this.url = url;
         this.protocol = new JsonHubProtocol();
         this.callback = (payload) -> {
 
+            if (!handshakeReceived) {
+                int handshakeLength = payload.indexOf(RECORD_SEPARATOR) + 1;
+                String handshakeResponseString = payload.substring(0, handshakeLength - 1);
+                HandshakeResponseMessage handshakeResponse = HandshakeProtocol.parseHandshakeResponse(handshakeResponseString);
+                if (handshakeResponse.error != null){
+                    throw new Exception("Error in handshake" + handshakeResponse.error);
```

Comment:
nit: space after "handshake", otherwise you'll get "Error in handshakeerror message" 😄

Reply:
This is why string interpolation is so much better ;P. #javajabs